File: src/main/scala/qmf/poc/agent/catalog/models/Catalog.scala
package qmf.poc.agent.catalog.models

import spray.json.{DefaultJsonProtocol, JsArray, JsNumber, JsObject, JsString, JsValue, JsonFormat, enrichAny, given}

class Catalog(val objectData: Seq[ObjectData], val objectRemarks: Seq[ObjectRemarks], val objectDirectories: Seq[ObjectDirectory])

object Catalog:
  given JsonFormat[Catalog] with
    def write(o: Catalog): JsObject =
      JsObject(
        "objectData" -> JsArray(o.objectData.map(_.toJson).toVector),
        "objectRemarks" -> JsArray(o.objectRemarks.map(_.toJson).toVector),
        "objectDirectories" -> JsArray(o.objectDirectories.map(_.toJson).toVector)
      )

    //noinspection NotImplementedCode
    def read(value: JsValue): Catalog = ???

File: src/main/java/qmf/poc/agent/catalog/models/Catalog.java
package qmf.poc.agent.catalog.models;

import java.util.List;
import java.util.stream.Collectors;

public class Catalog {
    public final List<QMFObject> qmfObjects;

    public Catalog(List<QMFObject> qmfObjects) {
        this.qmfObjects = qmfObjects;
    }

    @Override
    public String toString() {
        return "Catalog{" +
                "objects=" + qmfObjects.stream()
                .map(QMFObject::toString)
                .collect(Collectors.joining("\n")) +
                '}';
    }
}